plugin/pkg/shoot/validator/admission.go

```diff
@@ -1766,23 +1771,65 @@ func getDefaultMachineImage(machineImages []gardencorev1beta1.MachineImage, imag
 			}
 		}
 		if defaultImage == nil {
-			return nil, field.Invalid(fldPath, imageName, fmt.Sprintf("no valid machine image found that support architecture `%s`", *arch))
+			return nil, field.Invalid(fldPath, imageReference, fmt.Sprintf("no valid machine image found that supports architecture `%s`", *arch))
+		}
+	}
+
+	var (
+		machineImageVersionMajor *uint64
+		machineImageVersionMinor *uint64
+	)
+
+	if image != nil {
+		var err error
+		versionParts := strings.Split(strings.TrimPrefix(image.Version, "v"), ".")
+		if len(versionParts) == 3 {
+			return image, nil
+		}
+		if len(versionParts) == 2 && len(versionParts[1]) > 0 {
+			if machineImageVersionMinor, err = parseSemanticVersionPart(versionParts[1]); err != nil {
+				return nil, field.Invalid(fldPath, image.Version, err.Error())
+			}
+		}
+		if len(versionParts) >= 1 && len(versionParts[0]) > 0 {
+			if machineImageVersionMajor, err = parseSemanticVersionPart(versionParts[0]); err != nil {
+				return nil, field.Invalid(fldPath, image.Version, err.Error())
+			}
 		}
 	}
 
 	var validVersions []core.MachineImageVersion
 
 	for _, version := range defaultImage.Versions {
-		if slices.Contains(version.Architectures, *arch) {
-			validVersions = append(validVersions, version)
+		if !slices.Contains(version.Architectures, *arch) {
+			continue
+		}
+		// CloudProfile cannot contain invalid semVer machine image version
+		parsedVersion := semver.MustParse(version.Version)
+		if machineImageVersionMajor != nil && parsedVersion.Major() != *machineImageVersionMajor ||
+			machineImageVersionMinor != nil && parsedVersion.Minor() != *machineImageVersionMinor {
```

Comment:
`ptr.Deref`?

Reply:
I would still need to check for `machineImageVersionMajor != nil`, as else the `Deref` fallback is compared to the version part, leading to always continuing.

Or, do you mean this: `machineImageVersionMajor != nil && parsedVersion.Major() != ptr.Deref(machineImageVersionMajor, 0)`?

Reply:
as discussed: reverted, as it does not make sense here.